src/exo/LoopIR_scheduling.py

```diff
@@ -320,6 +321,12 @@ def DoReorderStmt(f_cursor, s_cursor):
     return ir, fwd
 
 
+def DoParallelizeLoop(loop_cursor):
```

Comment:
I still think parallel check shouldn't be a part of the rewriting process but should be a part of backend checks (same as memory and precision), because even if you check the safety in parallelize_loop , other scheduling primitives do not know that the loop is parallel and can rewrite the par loop to introduce the loop dependencies and break the safety.